test/functional/feature_evm_vmmap_rpc.py

```diff
@@ -94,65 +94,66 @@ def vmmap_valid_address_not_present_should_fail(self):
-        # for a block from the APIs and use that tx to test vmmap.
-        # That's the correct way. logvmmap and vmmap results in
-        # cyclic testing.
-        # list_tx = self.nodes[0].logvmmaps(1)
-        # dvm_tx = list(list_tx['indexes'].keys())[0]
-        # evm_tx = self.nodes[0].vmmap(dvm_tx, 3)
-        # assert_equal(dvm_tx, self.nodes[0].vmmap(evm_tx, 4))
-        # assert_equal("0x" + evm_tx, self.nodes[0].eth_getBlockByNumber("latest", False)['transactions'][0])
+        self.nodes[0].transferdomain([{"src": {"address": self.address, "amount": "100@DFI", "domain": 2}, "dst": {"address": self.ethAddress, "amount": "100@DFI", "domain": 3}}])
+        self.nodes[0].generate(1)
+        self.nodes[0].evmtx(self.ethAddress, 0, 21, 21000, self.toAddress, 1)
+        self.nodes[0].generate(1)
+        latest_tx = self.nodes[0].eth_getBlockByNumber("latest", False)['transactions'][0]
+        dvm_tx = self.nodes[0].vmmap(latest_tx, 4)
+        assert_equal(dvm_tx in self.nodes[0].getblock(self.nodes[0].getbestblockhash())['tx'], True)
 
     def vmmap_invalid_tx_should_fail(self):
         self.rollback_to(self.start_block_height)
         # Check vmmap fail on wrong tx
-        # TODO:
-        #   - Check for multiple ones.
-        #   - Get a valid tx, prefix it with garbage and check fail.
-        #   - Get a valid tx, suffix it with garbage and check fail.
+        latest_block = self.nodes[0].eth_getBlockByNumber("latest", False)['hash']
         fake_evm_tx = '0x0000000000000000000000000000000000000000000000000000000000000000'
         assert_raises_rpc_error(-32600, "Key not found: 0000000000000000000000000000000000000000000000000000000000000000", self.nodes[0].vmmap, fake_evm_tx, 4)
         assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, "0x00", 4)
         assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, "garbage", 4)
         assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, "0", 4)
         assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, "x", 4)
+        assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, latest_block + "0x00", 4)
+        assert_raises_rpc_error(-32600, None, self.nodes[0].vmmap, "0x00" + latest_block, 4)
 
     def vmmap_valid_block_should_succeed(self):
         self.rollback_to(self.start_block_height)
         # Check if xvmmap is working for Blocks
-        # TODO:
-        #   - Check for multiple ones. Generate and verify hashes manually.
-        #     Not have cyclic results
-        #   - Get a valid tx, prefix it with garbage and check fail.
-        #   - Get a valid tx, suffix it with garbage and check fail.
         latest_block = self.nodes[0].eth_getBlockByNumber("latest", False)
         dvm_block = self.nodes[0].vmmap(latest_block['hash'], 6)
-        assert_equal(latest_block['hash'], "0x" + self.nodes[0].vmmap(dvm_block, 5))
+        assert_equal(dvm_block, self.nodes[0].getbestblockhash())
```

Comment:
Could you loop through a set, and add 4-5 TXs

>        #   - Check for multiple ones. Generate and verify hashes manually.

Or, you just checking for a single item in the list.